recaptcha/recaptchalib.php

```diff
@@ -62,33 +64,17 @@ function _recaptcha_qsencode ($data) {
  * @param string $path
  * @param array $data
  * @param int port
- * @return array response
+ * @return string response
  */
 function _recaptcha_http_post($host, $path, $data, $port = 80) {
-
-        $req = _recaptcha_qsencode ($data);
-
-        $http_request  = "POST $path HTTP/1.0\r\n";
-        $http_request .= "Host: $host\r\n";
-        $http_request .= "Content-Type: application/x-www-form-urlencoded;\r\n";
-        $http_request .= "Content-Length: " . strlen($req) . "\r\n";
-        $http_request .= "User-Agent: reCAPTCHA/PHP\r\n";
-        $http_request .= "\r\n";
-        $http_request .= $req;
-
-        $response = '';
-        if( false == ( $fs = @fsockopen($host, $port, $errno, $errstr, 10) ) ) {
-                die ('Could not open socket');
-        }
-
-        fwrite($fs, $http_request);
-
-        while ( !feof($fs) )
-                $response .= fgets($fs, 1160); // One TCP-IP packet
-        fclose($fs);
-        $response = explode("\r\n\r\n", $response, 2);
-
-        return $response;
+  $client = new Client();
+  try {
+    $response = $client->request('POST', $host . '/' . $path, ['query' => $data]);
+  }
+  catch (Exception $e) {
+    return NULL;
```

Comment:
return NULL if it all fails?  Is this what we want?

Reply:
I've switched this to an empty string which I think is more consistent

Reply:
It's important to understand what's happening here.  A failure here would mean somehow being unable to connect to Google to verify the recaptcha.  Returning an empty string would cause the calling code to view it as a failure, therefore the user would be implored to re-enter the captcha.  It could be argued that if something is wrong with calling the verify server, then this isn't something that should bother the user and the transaction should go ahead.  The administrator needs to know though, and should be alerted that recaptcha verification isn't working.